src/wp-includes/class-wp-theme-json-resolver.php

```diff
@@ -145,35 +75,7 @@ private static function extract_paths_to_translate( $i18n_partial, $current_path
 	 * @return array An array of theme.json fields that are translatable and the keys that are translatable.
 	 */
 	public static function get_fields_to_translate() {
-		if ( null === self::$theme_json_i18n ) {
-			$file_structure        = self::read_json_file( __DIR__ . '/theme-i18n.json' );
-			self::$theme_json_i18n = self::extract_paths_to_translate( $file_structure );
-		}
-		return self::$theme_json_i18n;
-	}
-
-	/**
-	 * Translates a chunk of the loaded theme.json structure.
-	 *
-	 * @since 5.8.0
-	 *
-	 * @param array  $array_to_translate The chunk of theme.json to translate.
-	 * @param string $key                The key of the field that contains the string to translate.
-	 * @param string $context            The context to apply in the translation call.
-	 * @param string $domain             Text domain. Unique identifier for retrieving translated strings.
-	 * @return array Returns the modified $theme_json chunk.
-	 */
-	private static function translate_theme_json_chunk( array $array_to_translate, $key, $context, $domain ) {
-		foreach ( $array_to_translate as $item_key => $item_to_translate ) {
-			if ( empty( $item_to_translate[ $key ] ) ) {
-				continue;
-			}
-
-			// phpcs:ignore WordPress.WP.I18n.LowLevelTranslationFunction,WordPress.WP.I18n.NonSingularStringLiteralText,WordPress.WP.I18n.NonSingularStringLiteralContext,WordPress.WP.I18n.NonSingularStringLiteralDomain
-			$array_to_translate[ $item_key ][ $key ] = translate_with_gettext_context( $array_to_translate[ $item_key ][ $key ], $context, $domain );
-		}
-
-		return $array_to_translate;
+		return array();
```

Comment:
`get_fields_to_translate` became obsolete but it was marked as `public` so the question is whether we should keep all the logic with a deprecation notice or warn when someone tries to use it?

It seems like @schlessera was testing this method outside of WordPress core when he discovered the bug in the processing logic. That would be a good context for the decision here. I'm not sure if the method was really to be public in the first place or was it a consequence of having it exposed to cover it with a unit test.

Reply:
Ugh, too bad this was declared `public`.

+1 to a deprecation notice. I think an empty array is fine.

Reply:
I didn't directly use this when I discovered the bug. I reviewed this code as part of a PR to WP-CLI, where this logic was copied, and when I read over the code, I noticed it didn't make sense, but never produced a test failure as the test only tested the one case that actually worked.
So, in case that helps make a decision here, I did not rely in any way on this method being public.